ote/src/cljs/ote/views/own_services.cljs

```diff
@@ -103,23 +131,73 @@
            [delete-service-action e! row]]])
        services))])
 
-(defn transport-services-listing [e! transport-operator-id services section-label]
+(defn- route-error
+  [errors info-open?]
+  (let [error-keys (reduce
+                     (fn [set error]
+                       (conj set (:error error)))
+                     #{}
+                     errors)]
+    [info/info-toggle
+     (tr [:own-services-page :missing-service-info])
+     [:div
+      [:h5 {:style {:margin-top 0}} (tr [:own-services-page :flaws])]
+      (doall
+        (for [error errors]
+          ^{:key (:name error)}
+          [:p {:style {:margin 0}}
+           [:strong [ic/alert-warning {:style {:color colors/negative-button
+                                               :margin-bottom "-5px"}}] (:name error)]
+           " • " (tr [:own-services-page :missing-schedule])]))
+      [:h5 (tr [:own-services-page :guide-to-completion])]
+      [:span
+       [:strong (tr [:own-services-page :add-schedule-to-service]) " "]
+       (tr [:own-services-page :add-schedule-to-service-2]) " ("
+       [ic/content-create {:style {:margin-bottom "-5px"}}]
+       ") " (tr [:own-services-page :add-schedule-to-service-3])
+       [:strong " " (tr [:service-viewer :published-interfaces])]]
+      [:h5 (tr [:own-services-page :schedule-creation])]
+      [:p {:style {:margin-top 0}}
+       (tr [:own-services-page :if-no-interfaces])]
+      (when (error-keys :no-schedule-road)
```

Comment:
Generally when logic is only one case will be selected, then prefer
`case` condition instead of multiple `when`s. Reasoning: performance when not all are evaluated always, clarity when case tells reader only one case is selected

Reply:
But this doesn't mean that only one is selected. `error-keys` contain a set of errors that can contain both of the errors.